torch/csrc/Stream.cpp

```diff
@@ -256,6 +256,73 @@ static PyObject* THPStream_eq(THPStream* self, THPStream* other) {
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject* THPStream_enter(THPStream* self, PyObject* unused) {
+  HANDLE_TH_ERRORS
+  // No operation is performed if the stream does not belong to an accelerator.
+  c10::DeviceType dst_device_type =
+      static_cast<c10::DeviceType>(self->device_type);
+  if (C10_UNLIKELY(!at::accelerator::isAccelerator(dst_device_type))) {
+    Py_RETURN_NONE;
+  }
+  c10::DeviceIndex src_device_idx = at::accelerator::getDeviceIndex();
```

Comment:
Shouldn't the device type here be the one inside the stream and not the current accelerator?

Reply:
I added a `TORCH_CHECK` to ensure the device type inside the stream is equal to the current accelerator. And filter out `torch.Stream("cpu")` via the condition of `C10_UNLIKELY(!at::accelerator::isAccelerator(dst_device_type)`.

Reply:
I removed the `TORCH_CHECK` because `torch.Stream` can NOT be created if its device type (excluding CPU) is different from the current accelerator.